numpy/lib/arraypad.py

```diff
@@ -997,52 +996,75 @@ def _normalize_shape(narray, shape):
     ----------
     narray : ndarray
         Input ndarray
-    shape : {sequence, int}, optional
-        The width of padding (pad_width) or the number of elements on the
-        edge of the narray used for statistics (stat_length).
+    shape : {sequence, array_like, float, int}, optional
```

Comment:
I'm not happy about overloading `shape` this way, it is confusing. Would it make sense to have different functions for the various ways of padding? I'm not that happy with the orginal overloading either :0. Just fishing for alternative ideas here.